mypy/semanal.py

```diff
@@ -2120,14 +2120,17 @@ def analyze_namedtuple_assign(self, s: AssignmentStmt) -> bool:
         """Check if s defines a namedtuple."""
         if isinstance(s.rvalue, CallExpr) and isinstance(s.rvalue.analyzed, NamedTupleExpr):
             return True  # This is a valid and analyzed named tuple definition, nothing to do here.
-        if len(s.lvalues) != 1 or not isinstance(s.lvalues[0], NameExpr):
+        if len(s.lvalues) != 1 or type(s.lvalues[0]) not in [NameExpr, MemberExpr]:
             return False
         lvalue = s.lvalues[0]
         name = lvalue.name
```

Comment:
here a CI error is raised `"error: Expression" has no attribute "name" `,   
its confusing because lvalue is either NameExpr or MemberExpr both of which have `name` attribute.

Reply:
Mypy's type inference isn't smart enough to figure that out. You should probably move the `isinstance()` check to operate on the `lvalue` variable.

Reply:
I managed to get the CI green but the code isn't the best.